src/docker/shared.rs

```diff
@@ -1,8 +1,12 @@
+pub mod custom;
```

Comment:
Can't `custom` just be under `docker`? It can be it's own module and imported into shared. Not sure we need another level of nesting.